benchmark/test/reference/spmv_distributed.profile.stdout

```diff
@@ -5,7 +5,7 @@
         "comm_pattern": "stencil",
         "spmv": {
             "csr-csr": {
-                "storage": 6564,
+                "storage": 6420,
```

Comment:
Maybe not directly related to this, but I went back to check, there are some places left over in #1543 that still use std::vector instead of gko::vector or gko::array.

Reply:
I've only checked `(reference|omp)/distributed` for `std::vector` and found only 4 instances. Did you find more? (I'm updating them btw)

Reply:
no, I checked the patch, that should be all. The rest happens inside tests

Reply:
I see. Maybe also providing the maximal storage might be interesting to add in the future.

Reply:
Yes, good point!